devtools/gradle/gradle-model/src/main/java/io/quarkus/gradle/extension/ConfigurationUtils.java

```diff
@@ -0,0 +1,50 @@
+package io.quarkus.gradle.extension;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.provider.ListProperty;
+import org.gradle.api.provider.Property;
+import org.jetbrains.annotations.NotNull;
+
+// This is necessary because in included builds the returned `Project` instance and
+// `QuarkusExtensionConfiguration` extension is provided by a different class loader
+// which prevents us from creating some interface or casting to it directly.
+public class ConfigurationUtils {
+    private static final ConcurrentHashMap<String, Method> methodCache = new ConcurrentHashMap<>();
```

Comment:
Not sure whether a `static final` cache is good, because it would accumulate `Method` and with those `Class` and `ClassLoader` references. I suspect this would also not respect changes to the extensions and re-use old ones.

Reply:
Yea - or if needed, cache it for a build, but that's quite tricky w/ Gradle

Reply:
I removed the cache ;)